src/main/java/com/google/api/codegen/discovery/transformer/go/GoSampleMethodToViewTransformer.java

```diff
@@ -0,0 +1,216 @@
+import com.google.api.codegen.discovery.transformer.SampleMethodToViewTransformer;
+import com.google.api.codegen.discovery.transformer.SampleNamer;
+import com.google.api.codegen.discovery.transformer.SampleTransformerContext;
+import com.google.api.codegen.discovery.transformer.SampleTypeTable;
+import com.google.api.codegen.discovery.viewmodel.SampleAuthView;
+import com.google.api.codegen.discovery.viewmodel.SampleFieldView;
+import com.google.api.codegen.discovery.viewmodel.SamplePageStreamingView;
+import com.google.api.codegen.discovery.viewmodel.SampleView;
+import com.google.api.codegen.util.Name;
+import com.google.api.codegen.util.SymbolTable;
+import com.google.api.codegen.util.go.GoTypeTable;
+import com.google.api.codegen.viewmodel.ViewModel;
+import com.google.protobuf.Method;
+
+public class GoSampleMethodToViewTransformer implements SampleMethodToViewTransformer {
+
+  private final static String TEMPLATE_FILENAME = "go/sample.snip";
+
+  public GoSampleMethodToViewTransformer() {}
+
+  @Override
+  public ViewModel transform(Method method, SampleConfig sampleConfig) {
+    SampleTypeTable sampleTypeTable =
+        new SampleTypeTable(
+            new GoTypeTable(), new GoSampleTypeNameConverter(sampleConfig.packagePrefix()));
+    GoSampleNamer goSampleNamer = new GoSampleNamer();
+    SampleTransformerContext context =
+        SampleTransformerContext.create(
+            sampleConfig, sampleTypeTable, goSampleNamer, method.getName());
+    return createSampleView(context);
+  }
+
+  private SampleView createSampleView(SampleTransformerContext context) {
+    SampleConfig config = context.getSampleConfig();
+    MethodInfo methodInfo = config.methods().get(context.getMethodName());
+    SampleNamer namer = context.getSampleNamer();
+    SampleTypeTable typeTable = context.getSampleTypeTable();
+    SymbolTable symbolTable = SymbolTable.fromSeed(GoTypeTable.RESERVED_IDENTIFIER_SET);
+    addStaticImports(context, symbolTable);
+
+    SampleView.Builder builder = SampleView.newBuilder();
+
+    String clientVarName = symbolTable.getNewSymbol(namer.localVarName(Name.lowerCamel("c")));
+
+    String servicePackageName = GoSampleNamer.getServicePackageName(config.packagePrefix());
+    String serviceVarName = symbolTable.getNewSymbol(namer.getServiceVarName(servicePackageName));
+    List<String> methodNameComponents = new ArrayList<String>();
+    for (String nameComponent : methodInfo.nameComponents()) {
+      methodNameComponents.add(namer.publicFieldName(Name.lowerCamel(nameComponent)));
+    }
+    String requestVarName = symbolTable.getNewSymbol(namer.localVarName(Name.lowerCamel("req")));
+    // For this and other type name assignments, we don't go through the symbol
+    // table for some minor conveniences in the sample template. Since all types
+    // we care about are imported with the package in addStaticImports we don't
+    // have to worry about missing imports.
```

Comment:
I'm not clear what this comment means:

* what are the "minor conveniences" you refer to?
* is the second sentence related to the first?